docs/datastorage/datafixers.md

```diff
@@ -0,0 +1,107 @@
+The DataFixer System
+====================
+
+The DataFixer system is provided by vanilla, and allows for fine-grained migrations and adjustments to the world format. Forge expands this system to allow mods to register and use their own fixers.
+
+What would I use the DataFixer for?
+-----------------------------------
+Here are some things vanilla uses datafixers for:
+
+* Renaming the old `HealF` health tag to `Health` (1.9)
+* Replacing all integer ID's in ItemStacks with registry names (~ 1.9)
+* Renaming the ID's of all entities and tile entities from UpperCamel names like `Blaze` to proper registry names like `minecraft:blaze` (1.11)
+* Splitting entities that were variants of each other to separate ID's, e.g. `Elder` tag on `minecraft:guardian` -> `minecraft:elder_guardian` (1.11)
+* Flattening all meta-variants of an Item into separate ID's, e.g. `minecraft:wool` meta 0 -> `minecraft:white_wool` (1.13)
+
+As you can see, this can be used in a very powerful manner to implement any NBT save format fixes you would like to make, without resorting to hacks such as checking for the old tag every tick and replacing it with a new tag.
```

Comment:
Does this also cover the remapping done through the FMLMissingMappingsEvent?

Reply:
no, FMLMissingMappingsEvent is a completely orthogonal mechanism that rewrites entries in the FML ID maps, instead of fixing the world data. So the old name would point to the new item in the registry, making the old name still usable.

Reply:
sounds good, could we clarify that in the page?